core/deployment/src/main/java/io/quarkus/deployment/logging/LoggingResourceProcessor.java

```diff
@@ -209,6 +209,7 @@ LoggingSetupBuildItem setupLoggingRuntimeInit(LoggingSetupRecorder recorder, Log
                     possibleSupplier, launchModeBuildItem.getLaunchMode());
             LogConfig logConfig = new LogConfig();
             ConfigInstantiator.handleObject(logConfig);
+            logConfig.categories.clear(); // FIXME: workaround for duplicate handlers in CategoryConfiguredHandlerTest:42
```

Comment:
@stuartwdouglas @geoand this where I need some help:
The call on the previous line will now also provide category config (as expected). This makes the test fail because it only expected one ConsoleHandler and one FileHandler, not two of each.

I must say I'm very confused why `LogConfig` is recreated here (it's already fully availble to this method via `log`, e.g. it's passed to `recorder.initializeLogging()` on line 206).
I also fail to understand how `LoggingSetupRecorder.initializeBuildTimeLogging()` (called further down) was ever able to handle category config: https://github.com/quarkusio/quarkus/blob/2b4496718ec46836e2775f669c7c8c5f970ccaa1/core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java#L263
I mean, before my change `Maps` weren't even evaluated/populated...?

I'm less than sure what's the right approach here. Is the test to be adjusted? But aren't duplicated handlers a problem indeed?

Reply:
To be honest, I can't say I know the reason either

Reply:
@stuartwdouglas can you shed some light on this?

Reply:
> I think you need to add a clearHandlers() call to io.quarkus.bootstrap.logging.QuarkusDelayedHandler#buildTimeComplete.

Makes sense and it does work! Thanks!

> The reason why LogConfig is created here is because it is runtime config, and this is a build time method. It's kinda hacky, but we basically just use whatever the current config would be to initialize build time logging (just so we don't have to duplicate all logging config into a new build time config).

Ok, so `LogConfig log` passed to this method is not to be confused with `LogConfig logConfig`?